### PR TITLE
Refactor dds gateway's configuration part.

### PR DESCRIPTION
**Pull request by alvin1221**

### CODE REVIEW

**Comment by alvin1221** [comment]
pls merge nng first https://github.com/nanomq/NanoNNG/pull/407

**Review by lee-emqx — APPROVED** [review]

File: etc/nanomq_dds_gateway.conf [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        # # Value: Enum
+        # # - 5: mqttv5
+        # # - 4: mqttv311
+        # # - 3: mqttv31
+        proto_ver = 4
+        # # The ClientId of a remote bridge.
+        # # Default random string.
+        # #
+        # # Value: String
+        # clientid="bridge_client"
```
Comment (wanghaEMQ) [inline etc/nanomq_dds_gateway.conf:31]: Is this clientid should be commented?

File: etc/nanomq_dds_gateway.conf [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            # # This certificate is used to identify the AWS IoT server and is publicly
+            # # available.
+            # #
+            # # Value: File
+            cacertfile = "/etc/certs/cacert.pem"
+        }
+    }
+}
+
+forward_rules = {
```
Comment (wanghaEMQ) [inline etc/nanomq_dds_gateway.conf:88]: forward rules should better be shown first

File: etc/nanomq_dds_gateway.conf (hunk 1 above)
Comment (alvin1221) [reply etc/nanomq_dds_gateway.conf:31]: Yes, default random string;

File: etc/nanomq_dds_gateway.conf (hunk 2 above)
Comment (alvin1221) [reply etc/nanomq_dds_gateway.conf:88]: move it to the top of file?

File: nanomq_cli/dds2mqtt/mqtt_client.h [hunk 3]
```diff
@@ -39,21 +40,23 @@ struct mqtt_cli {
 
 	char *mqttrecv_topic;
 	char *mqttsend_topic;
+
+	dds_gateway_conf *config;
 };
 
-int mqtt_connect(mqtt_cli *cli, const char *url, void *ddscli);
+extern int mqtt_connect(mqtt_cli *cli, void *ddscli, dds_gateway_conf *config);
```
Comment (wanghaEMQ) [inline nanomq_cli/dds2mqtt/mqtt_client.h:47]: No extern to avoid misunderstanding.

File: etc/nanomq_dds_gateway.conf (hunk 2 above)
Comment (wanghaEMQ) [reply etc/nanomq_dds_gateway.conf:88]: Yes

**Comment by wanghaEMQ** [comment]
Okey. Good job

**Review by wanghaEMQ — APPROVED** [review]

**Comment by JaylinYu** [comment]
conflict with #1001

**Comment by alvin1221** [comment]
> conflict with #1001

Emmm, but I have implemented cmd parser by `nng_opts_parse()`

File: etc/nanomq_dds_gateway.conf [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    mqtt_to_dds =
+    {
+        from_mqtt = "DDSCMD/topic1"
+        to_dds = "MQTT/topic1"
+    }
+}
+
+dds {
+	idl_type  = topic1Type
```
Comment (JaylinYu) [inline etc/nanomq_dds_gateway.conf:20]: struct type?

**Review by JaylinYu — APPROVED** [review]

**Pull request merged**